Source/WebCore/html/HTMLButtonElement.cpp

```diff
@@ -155,10 +155,15 @@ void HTMLButtonElement::defaultEventHandler(Event& event)
 
             if (m_type == SUBMIT || m_type == RESET)
                 event.setDefaultHandled();
-        } else if (invokeTargetElement()) {
-            handleInvokeAction();
-        } else
-            handlePopoverTargetAction();
+        }
+
+        if (!protectedForm || m_type == BUTTON) {
```

Comment:
Marking as draft because this maybe should allow reset buttons too.

Reply:
Does the WPT cover this and answer that question? If not, can we add a test rather than just deciding based on our thinking?

Reply:
Apologies I should have resolved this comment. The spec (and Firefox) both support reset buttons so I've made them work in WebKit and updated the wpt to match.